virttest/utils_test/qemu/__init__.py

```diff
@@ -657,27 +657,23 @@ def hotplug_memory(self, vm, name):
     def unplug_memory(self, vm, name):
         """
         Unplug memory device
-        step 1, unplug dimm device
+        step 1, unplug the associated dimm or virtio-mem device
         step 2, unplug memory object
 
         :param vm: VM object
         :param name: memory device name
         """
         devices = []
-        qid_dimm = "dimm-%s" % name
+        qid_list = ["dimm-%s" % name, "virtio_mem-%s" % name]
         qid_mem = "mem-%s" % name
-        try:
-            dimm = vm.devices.get_by_qid(qid_dimm)[0]
-        except IndexError:
-            LOG.warn("'%s' is not used by any dimm" % qid_mem)
-        else:
-            step = "Unplug pc-dimm '%s'" % qid_dimm
-            error_context.context(step, LOG.info)
-            _, ver_out = vm.devices.simple_unplug(dimm, vm.monitor)
-            if ver_out is False:
-                raise exceptions.TestFail("Verify unplug memory failed")
-            devices.append(dimm)
-            self.update_vm_after_unplug(vm, dimm)
+
+        for qid in qid_list:
+            try:
+                device = vm.devices.get_by_qid(qid)[0]
+            except IndexError:
+                LOG.warn("'%s' is not used by %s" % (qid_mem, qid))
+            else:
+                self.device_unplug(device, qid, devices, vm)
```

Comment:
```suggestion
        for qid in qid_list:
            try:
                device = vm.devices.get_by_qid(qid)[0]
            except IndexError:
                pass
            else:
                self.device_unplug(device, qid, devices, vm)
                break
        else:
            LOG.warn("'%s' is not used any memory devices", qid_mem)
```

as

1. need to end the loop once there was any target dev node been found and unplugged
2. report the warning (or raise exceptions instead) only if the given memory object not associated with any memory device

Does that make sense?

Reply:
@luckyh yes, good suggestion ! Thanks :grinning:

Reply:
Code updated !